ansible/tasks/check_if_should_build.yaml

```diff
@@ -29,8 +29,8 @@
   - name: Setting Facts from Build
     set_fact:
       # if a build is cancelled the commit field is undefined inside of the build
-      build_commit: "{% if build.resources[0].spec.revision is defined and build.resources[0].spec.revision.git.commit is defined  %}{{ build.resources[0].spec.revision.git.commit }}{% else %}''{% endif %}"
-      build_phase: "{{ build.resources[0].status.phase }}"
+      build_commit: "{% if build is defined and build.resources is defined and build.resources[0] is defined and build.resources[0].spec is defined and build.resources[0].spec.revision is defined and build.resources[0].spec.revision.git.commit is defined  %}{{ build.resources[0].spec.revision.git.commit }}{% else %}''{% endif %}"
```

Comment:
:) I am new to ansible development as well. Thanks for the link. I will take a look and update accordingly.

Reply:
Just checked the link you mentioned. I am not sure if it is applicable here at all. Given the null reference error happens somewhere when we access .spec.revision (which is internal instead of being the last variable), this wouldn't help us here.

Reply:
Okay cool, I was hoping that the null reference error would cause the default value to be set.